modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/FolderActionDisplayContext.java

```diff
@@ -75,6 +85,168 @@ public FolderActionDisplayContext(
+							LanguageUtil.get(_httpServletRequest, "move"));
+					}
+				).add(
+					() -> isDeleteExpiredTemporaryFileEntriesActionVisible(),
+					dropdownItem -> {
+						dropdownItem.setHref(
+							getDeleteExpiredTemporaryFileEntriesURL());
+						dropdownItem.setLabel(
+							LanguageUtil.get(
+								_httpServletRequest,
+								"delete-expired-temporary-files"));
+					}
+				).add(
+					() -> isAddFolderActionVisible(),
+					dropdownItem -> {
+						dropdownItem.setHref(getAddFolderURL());
+						dropdownItem.setLabel(
+							LanguageUtil.get(
+								_httpServletRequest, "add-folder"));
+					}
+				).add(
+					() -> isAddRepositoryActionVisible(),
+					dropdownItem -> {
+						dropdownItem.setHref(getAddRepositoryURL());
+						dropdownItem.setLabel(
+							LanguageUtil.get(
+								_httpServletRequest, "add-repository"));
+					}
+				).add(
+					() -> isAddMediaActionVisible(),
+					dropdownItem -> {
+						dropdownItem.setHref(getAddMediaURL());
+						dropdownItem.setLabel(
+							LanguageUtil.get(
+								_httpServletRequest, "add-file-entry"));
+					}
+				).add(
+					() ->
+						isAddMediaActionVisible() &&
+						isMultipleUploadSupported(),
+					dropdownItem -> {
+						dropdownItem.put(
+							"class",
+							"dropdown-item hide upload-multiple-documents");
+						dropdownItem.setHref(getAddMultipleMediaURL());
+						dropdownItem.setLabel(
+							LanguageUtil.get(
+								_httpServletRequest, "multiple-media"));
+					}
+				).add(
+					() -> isViewSlideShowActionVisible(),
+					dropdownItem -> {
+						dropdownItem.putData("action", "slideShow");
+						dropdownItem.putData(
+							"viewSlideShowURL", getViewSlideShowURL());
+						dropdownItem.setLabel(
+							LanguageUtil.get(
+								_httpServletRequest, "view-slide-show"));
+					}
+				).add(
+					() -> isAddFileShortcutActionVisible(),
+					dropdownItem -> {
+						dropdownItem.setHref(getAddFileShortcutURL());
+						dropdownItem.setLabel(
+							LanguageUtil.get(
+								_httpServletRequest, "add-shortcut"));
+					}
+				).add(
+					() -> isAccessFromDesktopActionVisible(),
+					dropdownItem -> {
+						dropdownItem.putData("action", "accessFromDesktop");
+
+						LearnMessage learnMessage =
+							LearnMessageUtil.getLearnMessage(
+								"webdav",
+								LanguageUtil.getLanguageId(_httpServletRequest),
+								"document-library-web");
+
+						dropdownItem.putData(
+							"learnMessage", learnMessage.getMessage());
+						dropdownItem.putData(
+							"learnURL", learnMessage.getMessage());
+
+						ThemeDisplay themeDisplay =
+							(ThemeDisplay)_httpServletRequest.getAttribute(
+								WebKeys.THEME_DISPLAY);
+
+						dropdownItem.putData(
+							"webDavURL",
+							DLURLHelperUtil.getWebDavURL(
+								themeDisplay, _getFolder(), null));
+
+						dropdownItem.setLabel(
+							LanguageUtil.get(
+								_httpServletRequest, "access-from-desktop"));
+					}
+				).add(
+					() -> isPermissionsActionVisible(),
```

Comment:
maybe is worth to make all the `isPermissionsActionVisible` ... `isDeleteFolderActionVisible` private now that they are not called from the jsp?

Reply:
It is a good point, I can do it, if you want